File: leet_code/trees/range_sum.rb
def range_sum_bst(root, l, r)
    if (!root)
        return
    else
        sum = 0
        traverse = lambda do |node|
            traverse.call(node.left) if node.left
            sum += node.val if node.val >= l && node.val <= r
            traverse.call(node.right) if node.right
        end
        traverse.call(root)
        return sum
    end
end

File: leet_code/trees/range_sum.js
const rangeSumBST = (root, L, R) => {
  if (!root) return;
  else {
    let sum = 0;
    const traverse = node => {
      if (node.left) traverse(node.left);
      if (node.val >= L && node.val <= R) sum += node.val;
      if (node.right) traverse(node.right);
    };
    traverse(root);
    return sum;
  }
};
